app/screens/send-bitcoin-screen/send-lnurl-screen.tsx

```diff
@@ -0,0 +1,323 @@
+import {
+  invoiceRequest,
+  isAmountValid,
+  isProtocolSupported,
+  parseUrl,
+} from "../../utils/lnurl"
+import { mSatToSat } from "../../utils/amount"
+import { useMoneyAmount, useBTCPrice, useCurrencies } from "../../hooks"
+
+const PAY_INVOICE = gql`
+  mutation payInvoice($invoice: String!, $amount: Int, $memo: String) {
+    invoice {
+      payInvoice(invoice: $invoice, amount: $amount, memo: $memo)
+    }
+  }
+`
+
+const styles = EStyleSheet.create({
+  animationContainer: { alignItems: "center" },
+  buttonContainerStyle: {
+    flex: 1,
+  },
+  buttonStyle: {
+    backgroundColor: color.primary,
+    marginBottom: 32,
+    marginHorizontal: 24,
+    marginTop: 32,
+  },
+  description: {
+    marginTop: 10,
+  },
+  errorContainer: {
+    alignItems: "center",
+    flex: 1,
+    justifyContent: "center",
+  },
+  errorText: {
+    color: palette.red,
+    fontSize: 18,
+  },
+  lottie: {
+    height: "200rem",
+    width: "200rem",
+  },
+  memoContainer: {
+    marginLeft: "-11rem",
+    marginVertical: 12,
+  },
+  memoInput: {
+    fontSize: 14,
+  },
+  memoLabel: {
+    color: palette.midGrey,
+    fontSize: 14,
+    fontWeight: "normal",
+    marginBottom: "3rem",
+    marginLeft: "1rem",
+  },
+
+  scroll: {
+    justifyContent: "space-between",
+  },
+  section: {
+    alignItems: "center",
+    marginHorizontal: 48,
+  },
+
+  successText: {
+    fontSize: 18,
+  },
+
+  transactionDetailView: {
+    alignItems: "center",
+    flex: 1,
+    justifyContent: "center",
+    marginHorizontal: "24rem",
+    marginVertical: "24rem",
+  },
+})
+
+type Props = {
+  route: {
+    params: {
+      invoice: string
+    }
+  }
+}
+
+const rowStyles = EStyleSheet.create({
+  description: {
+    flexDirection: "row",
+    flexWrap: "wrap",
+    marginVertical: 12,
+    minWidth: "48%",
+  },
+
+  entry: {
+    color: palette.midGrey,
+    marginBottom: "6rem",
+  },
+
+  value: {
+    color: palette.darkGrey,
+    fontSize: "14rem",
+    fontWeight: "bold",
+  },
+})
+
+type RowProps = {
+  entry: string
+  value: string
+}
+
+export const Row = ({ entry, value }: RowProps): JSX.Element => {
+  return (
+    <View style={rowStyles.description}>
+      <Text style={rowStyles.entry}>{entry}</Text>
+      <Text selectable style={rowStyles.value}>
+        {value}
+      </Text>
+    </View>
+  )
+}
+
+export const SendLNUrlScreen: React.FC<Props> = ({ route }: Props) => {
+  const { navigate } = useNavigation()
+  const client = useApolloClient()
+  const balance = balanceBtc(client)
+
+  const btcPrice = useBTCPrice()
+  const { primaryCurrency, secondaryCurrency, toggleCurrency } = useCurrencies()
+  const [primaryAmount, convertPrimaryAmount, setPrimaryAmount, setPrimaryAmountValue] =
+    useMoneyAmount(secondaryCurrency)
+
+  const [
+    secondaryAmount,
+    convertSecondaryAmount,
+    setSecondaryAmount,
+    setSecondaryAmountValue,
+  ] = useMoneyAmount(secondaryCurrency)
+
+  const [amount, setAmount] = useState(0)
```

Comment:
This shouldn't be necessary with the useMoneyAmount. You can store needed amounts in there.